### PR TITLE
[INLONG-9897][Common] Code cognitive complexity modification

### PR DESCRIPTION
**Pull request by beatCao**
Title Name: [INLONG-9897] Code cognitive complexity modification

https://github.com/apache/inlong/issues/9897

Motivation
Explain here the context, and why you're making that change. What is the problem you're trying to solve.

Modifications
Describe the modifications you've done.

### CODE REVIEW

File: inlong-common/src/main/java/org/apache/inlong/common/monitor/CounterGroup.java [hunk 1]
```diff
@@ -59,9 +59,7 @@ public synchronized void set(String name, Long value) {
     }
 
     public synchronized AtomicLong getCounter(String name) {
-        if (!this.counters.containsKey(name)) {
-            this.counters.put(name, new AtomicLong());
-        }
+        this.counters.computeIfAbsent(name, k -> new AtomicLong());
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/monitor/CounterGroup.java:62]: Suggestion: 
```java
return this.counters.computeIfAbsent(name, k -> new AtomicLong());
```

File: inlong-common/src/main/java/org/apache/inlong/common/monitor/CounterGroupExt.java [hunk 2]
```diff
@@ -75,9 +75,7 @@ public synchronized void set(String name, Long value) {
     }
 
     public synchronized AtomicLong getCounter(String name) {
-        if (!this.counters.containsKey(name)) {
-            this.counters.put(name, new AtomicLong());
-        }
+        this.counters.computeIfAbsent(name, k -> new AtomicLong());
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/monitor/CounterGroupExt.java:78]: ditto here.

File: inlong-common/src/main/java/org/apache/inlong/common/monitor/CounterGroup.java (hunk 1 above)
Comment (beatCao) [reply inlong-common/src/main/java/org/apache/inlong/common/monitor/CounterGroup.java]: > Suggestion:
> 
> ```java
> return this.counters.computeIfAbsent(name, k -> new AtomicLong());
> ```

Problem fixed

File: inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsgAttrBuilder.java [hunk 3]
```diff
@@ -257,41 +259,48 @@ public String buildAttr() throws Exception {
...
-                        tstr = f.format(d) + "q" + idx;
-                    } else if (this.p == PartitionUnit.TENMINS) {
-                        int idx =
-                                (int) ((d.getTime() % (60L * 60 * 1000)) / (10L * 60 * 1000));
-                        SimpleDateFormat f = new SimpleDateFormat("yyyyMMddHH");
-                        tstr = f.format(d) + "t" + idx;
-                    } else if (this.p == PartitionUnit.FIVEMINS) {
-                        int idx =
-                                (int) ((d.getTime() % (60L * 60 * 1000)) / (5L * 60 * 1000));
+                    if (d != null) {
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsgAttrBuilder.java:269]: What if d is null?

File: inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsgAttrBuilder.java [hunk 4]
```diff
@@ -162,17 +162,19 @@ public String buildAttr() throws Exception {
...
-                tstr = f.format(d);
-            } else if (this.p == PartitionUnit.HOUR) {
-                SimpleDateFormat f = new SimpleDateFormat("yyyyMMddHH");
-                tstr = f.format(d);
-            } else if (this.p == PartitionUnit.QUARTER) {
-                int idx =
-                        (int) ((d.getTime() % (60L * 60 * 1000)) / (15L * 60 * 1000));
-                SimpleDateFormat f = new SimpleDateFormat("yyyyMMddHH");
-                tstr = f.format(d) + "q" + idx;
+            if (d != null) {
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsgAttrBuilder.java:165]: take care of `d==null`

File: inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsgAttrBuilder.java [hunk 5]
```diff
@@ -257,41 +259,48 @@ public String buildAttr() throws Exception {
...
+                                tstr = f.format(d) + "t" + idx;
+                            }
+                                break;
+                            case FIVEMINS: {
+                                int idx =
+                                        (int) ((d.getTime() % (60L * 60 * 1000)) / (5L * 60 * 1000));
+                                tstr = f.format(d) + "f" + idx;
+                            }
+                                break;
+                        }
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/msg/InLongMsgAttrBuilder.java:303]:
```suggestion
switch (this.p) {
                case DAY:
                    tstr = fd.format(d);
                    break;
                case HOUR:
                    tstr = f.format(d);
                    break;
                case HALFHOUR:
                    int idx = (int) ((d.getTime() % (60L * 60 * 1000)) / (30L * 60 * 1000));
                    tstr = f.format(d) + "n" + idx;
                    break;
                case QUARTER:
                    idx = (int) ((d.getTime() % (60L * 60 * 1000)) / (15L * 60 * 1000));
                    tstr = f.format(d) + "q" + idx;
                    break;
                case TENMINS:
                    idx = (int) ((d.getTime() % (60L * 60 * 1000)) / (10L * 60 * 1000));
                    tstr = f.format(d) + "t" + idx;
                    break;
                case FIVEMINS:
                    idx = (int) ((d.getTime() % (60L * 60 * 1000)) / (5L * 60 * 1000));
                    tstr = f.format(d) + "f" + idx;
                    break;
            }
```

**Comment by aloyszhang** [comment]
FYI, as I mentioned in #9896 https://github.com/apache/inlong/pull/9896#issuecomment-2026529790.

If the pull request is code refactoring or optimization, it's better to create an umbrella issue to track all the code optimization issues at the module or sub-module level, each module or sub-module has only one issue to deal with the code optimization.

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**